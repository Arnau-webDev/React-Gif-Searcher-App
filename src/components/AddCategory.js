import React, {useState} from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories, categories }) => {

const [inputValue, setInputValue] = useState("");

const handleInputChange = (e) => {
    setInputValue(e.target.value);
}

const handleSubmit = (e) => {
    e.preventDefault();

    categories.forEach((element, index) => {
        if(element.toUpperCase() === inputValue.toUpperCase()) {
            categories.splice(index, 1);
        }
    })
    
    if(inputValue.trim().length > 1 && inputValue.trim().length < 35 ) {
        setCategories(( categories ) => [inputValue, ...categories]);
        setInputValue("");
    }

    setInputValue("");
}

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                placeholder="Type about something..."
                value= { inputValue }
                onChange= { handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
