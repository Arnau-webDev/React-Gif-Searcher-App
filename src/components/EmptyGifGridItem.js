import React from 'react'

export const EmptyGifGridItem = ({ category }) => {

    return (
        <div className="notFoundCard animate__animated animate__fadeIn">
            <p>No GIFS where found from the <span>{ category }</span> search</p>
        </div>
    )
}
