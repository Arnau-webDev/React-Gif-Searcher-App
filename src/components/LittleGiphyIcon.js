import React from 'react';

export const LittleGiphyIcon = () => {

    const littleIconUrl = "https://media0.giphy.com/media/3xz2BDFvxop2BfAQoM/100w.gif?cid=6f54fedeoerrqrxeblu5mj93k2ryzyd366dbj9sca30ajebk&rid=100w.gif";

    return (
        <div>
            <div className="smallContainer">
                <div className="small animate__animated animate__backInDown">
                    <a href="https://developers.giphy.com/" alt="" target="_blank" rel="noopener noreferrer">
                        <div>Powered by GIPHY</div>
                        <div className="extraSmall"><img src={ littleIconUrl } alt=""></img></div>
                    </a>
                </div>
            </div>
            <h2>Gif Expert App</h2>
        </div>
    )
}
