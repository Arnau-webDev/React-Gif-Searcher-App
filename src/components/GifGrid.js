import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { EmptyGifGridItem } from './EmptyGifGridItem';

export const GifGrid = ({ category }) => {

    // const categoriesAlreadySearched = document.querySelectorAll("h3");

    //  categoriesAlreadySearched.forEach((element) => {
    //      console.log(element.innerHTML)
    //     if(element.innerHTML.toUpperCase() === category.trim().toUpperCase() ){
    //         return <></>
    //     }
        
    //  }); 

    const { data:images, loading, dataLength } = useFetchGifs( category );

    return (
        <div>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            { dataLength < 1 && loading === false && <EmptyGifGridItem category={ category } /> }

            <div className="card-grid">
                { images.map(( img ) => <GifGridItem key={ img.id } img={ img } /> )}
            </div>
        </div>
    )
}
