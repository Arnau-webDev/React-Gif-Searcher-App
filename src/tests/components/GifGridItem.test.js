import React from 'react';
import { GifGridItem } from "../../components/GifGridItem";
const { shallow } = require("enzyme")

describe('Pruebas GifGridItem', () => {

    const testImg = {
        id: "3o7budMRwZvNGJ3pyE",
        title: "baby groot hello GIF",
        url: "https://media0.giphy.com/media/3o7budMRwZvNGJ3pyE/200.gif?cid=6f54fede8s54pr16kgr6lf8j3r09deuycwk3wi9y69gfz6rg&rid=200.gif",
    }
    const wrapper = shallow( <GifGridItem img={ testImg } />);

    test('se muestra el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();

    })

    test("Se debe mostrar el titulo que se paso por props en el componente", () => {
        const value = wrapper.find("p").text().trim();
        
        expect( value ).toBe( testImg.title);
    })

    test("Se debe mostrar la imagen que se paso por props en el componente", () => {
        const img = wrapper.find("img");
        const imgAlt = img.props().alt;
        const imgUrl = img.props().src;
        
        expect( imgAlt ).toBe( testImg.title);
        expect( imgUrl ).toBe( testImg.url);
    })

    test("Debe de contener la clase animate__backInLeft", () => {
        const div = wrapper.find("div");
        const divClasses = div.props();
        const animatedClass = "animate__backInLeft";

        expect( divClasses.className.includes ( animatedClass ) ).toBe( true );

        console.log(divClasses);
    })
    
})
