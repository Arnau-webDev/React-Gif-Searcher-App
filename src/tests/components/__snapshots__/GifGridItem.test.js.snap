// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem se muestra el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="baby groot hello GIF"
    src="https://media0.giphy.com/media/3o7budMRwZvNGJ3pyE/200.gif?cid=6f54fede8s54pr16kgr6lf8j3r09deuycwk3wi9y69gfz6rg&rid=200.gif"
  />
  <p>
    baby groot hello GIF
  </p>
</div>
`;
